src/hotspot/share/utilities/globalDefinitions.hpp

```diff
@@ -90,72 +90,89 @@ class oopDesc;
 // doesn't provide appropriate definitions, they should be provided in
 // the compiler-specific definitions file (e.g., globalDefinitions_gcc.hpp)
 
-#define BOOL_TO_STR(_b_) ((_b_) ? "true" : "false")
+// Guide to the suffixes used in the format specifiers for integers:
+//        - print the decimal value:                   745565
+//  _X    - print as hexadecimal, without leading 0s: 0x12345
+//  _X_0  - print as hexadecimal, with leading 0s: 0x00012345
+//  _H    - print as hexadecimal, without 0x prefix
+//  _W(w) - prints w sized string with the given value right
+//          adjusted. Use -w to print left adjusted.
+//
+// Note that the PTR format specifiers print using 0x with leading zeros,
+// just like the _X_0 version for integers.
 
-// Format 32-bit quantities.
-#define INT32_FORMAT           "%" PRId32
-#define UINT32_FORMAT          "%" PRIu32
-#define INT32_FORMAT_W(width)  "%" #width PRId32
-#define UINT32_FORMAT_W(width) "%" #width PRIu32
+// Format 8-bit quantities.
+#define INT8_FORMAT_X_0          "0x%02"      PRIx8
+#define UINT8_FORMAT_X_0         "0x%02"      PRIx8
 
-#define PTR32_FORMAT           "0x%08" PRIx32
-#define PTR32_FORMAT_W(width)  "0x%" #width PRIx32
+// Format 16-bit quantities.
+#define INT16_FORMAT_X_0         "0x%04"      PRIx16
+#define UINT16_FORMAT_X_0        "0x%04"      PRIx16
```

Comment:
I'm confused.  What's the trailing 0 ?  is it to print the "0x" part of the hex format?  And why wouldn't you want that?

Reply:
I tried to describe it above:
```
//  _X    - print as hexadecimal, without leading 0s: 0x12345
//  _X_0  - print as hexadecimal, with leading 0s: 0x00012345
```

It's about padding with 0s. So, that you get 0x00012345 instead of 0x12345. Do you have a suggestion on how to write that comment any clearer?

Reply:
They have not been discussed. I looked for a pragmatic solution to the problem that parts of the code have different formatting requirements, and then selected one naming convention that I liked. This can still be changed if we can agree on a another / better naming convention.

Reply:
I don't have a strong opinion, just offering a couple ideas.  I'm not sure the additional underscore between the modifiers is worthwhile.  I like having the "X" before the extension part.  I thought about "Z" (for zero) instead of "0", but think I prefer being more literal.  So my preference would be "XXX_FORMAT_X0" instead of "XXX_FORMAT_X_0".  But that's a pretty weak preference.

Reply:
An alternative could be to switch so that _X means 0x and zero padding, and then use another letter to denote the case where we don't want the leading zeros.

Reply:
INTPTR_FORMAT zero pads, right? and that seems to be more prevalent in the code.  So THING_FORMAT_X seems like it should also pad since that's usually(?) what we want.  So I like this idea.  Then thinking of a way to say not to pad is hard - how about THING_FORMAT_X_NZ (NZ for no zero).  Are there a lot of these?
Sorry for the drive by comments. If other people are fine, I'm okay with the names. I was just a bit surprised by the trailing zero on first read.

Reply:
I think the naming convention in the PR is as good as any other - it is something new that we have to get used to.

Reply:
It's fine with me whatever you've decided.